gpkit/nomials.py

```diff
@@ -145,27 +144,45 @@ def diff(self, wrt):
         Arguments
         ---------
         wrt (Variable):
-            Variable to take derivative with respect to
+        Variable to take derivative with respect to
 
         Returns
         -------
         Signomial (or Posynomial or Monomial)
         """
-        if wrt in self.varstrs:
-            wrt = self.varstrs[wrt]
-        elif not isinstance(wrt, VarKey):
-            wrt = wrt.varkey
-        exps, cs = diff(self, wrt)
-        return Signomial(exps, cs, require_positive=False)
+        deriv = super(Signomial, self).diff(wrt)
+        return Signomial(exps=deriv.exps, cs=deriv.cs)
```

Comment:
once we refactor (i.e. remove) all the ways to create a `Signomial`, I'm hoping that `Signomial(deriv)` will work here. (deriv is a NomialData).